test/e2e/install/install.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package install
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/k8stopologyawareschedwg/deployer/pkg/deployer/platform"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	nropv1alpha1 "github.com/openshift-kni/numaresources-operator/api/numaresourcesoperator/v1alpha1"
+	"github.com/openshift-kni/numaresources-operator/pkg/status"
+	e2eclient "github.com/openshift-kni/numaresources-operator/test/utils/clients"
+	"github.com/openshift-kni/numaresources-operator/test/utils/configuration"
+	"github.com/openshift-kni/numaresources-operator/test/utils/machineconfigpools"
+	"github.com/openshift-kni/numaresources-operator/test/utils/objects"
+)
+
+var _ = Describe("[Install]", func() {
+	var initialized bool
+
+	BeforeEach(func() {
+		if !initialized {
+			Expect(e2eclient.ClientsEnabled).To(BeTrue(), "failed to create runtime-controller client")
+
+		}
+		initialized = true
+	})
+
+	Context("with a running cluster with all the components", func() {
+		It("should perform overall deployment and verify the condition is reported as available", func() {
+			var matchLabels map[string]string
+
+			if configuration.Platform == platform.Kubernetes {
```

Comment:
This is fine that we're creating MCP resources in k8s platform, but what about the CRD itself?

Reply:
BTW this can be addressed on a separate PR, since this code was just copied.

Reply:
Its part of the deployment procedure, the same true regarding NRO CRD

Reply:
What deployment procedure? 
we're using kustomize, but I don't see MCP CRD here: https://github.com/openshift-kni/numaresources-operator/tree/main/config/crd

Reply:
https://github.com/openshift-kni/numaresources-operator/blob/939709fee7a1c4b58463cb66e353f723139c2c90/Makefile#L155